napari/_qt/containers/qt_axis_model.py

```diff
@@ -0,0 +1,171 @@
+from typing import Any, Iterable, Union
+
+from qtpy.QtCore import QModelIndex, Qt
+
+from napari._qt.containers.qt_list_model import QtListModel
+from napari.components import Dims
+from napari.utils.events import SelectableEventedList
+
+
+class AxisModel:
+    """View of an axis within a dims model.
+
+    The model keeps track of axis names and allows read / write
+    acces on the cossesponding rollable state of a Dims object.
+
+    Parameters
+    ----------
+    dims : napari.components.dims.Dims
+        Parent Dims object.
+    axis : int
+        Axis index.
+
+    Attributes
+    ----------
+    dims : napari.components.dims.Dims
+        Dimensions object modeling slicing and displaying.
+    axis : int
+        Axis index.
+    """
+
+    def __init__(self, dims: Dims, axis: int) -> None:
+        self.dims = dims
+        self.axis = axis
+
+    def __hash__(self) -> int:
+        return id(self)
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def __repr__(self) -> str:
+        return self.dims.axis_labels[self.axis]
+
+    def __eq__(self, other: Union[int, str]) -> bool:
```

Comment:
Optional: if you want to fix mypy, just annotate the argument with `object`. We're doing type checking in the body, so this describes the situation fine and no longer breaks the Liskov substitution principle (i.e. you can't specialize argument types).

```suggestion
    def __eq__(self, other: object) -> bool:
```